src/server/transaction.cc

```diff
@@ -994,6 +961,11 @@ void Transaction::ExecuteAsync() {
   });
 }
 
+void Transaction::FinishHop() {
+  boost::intrusive_ptr<Transaction> guard(this);
```

Comment:
could be nice to add hop_index_ for debugging.

Reply:
To shard stats?

Reply:
no, for example to use it in Debug_FailedState or in general to have it in case we need to debug it. not for prod